src/overlays/actors/ovl_En_Encount3/z_en_encount3.c

```diff
@@ -32,22 +34,173 @@ ActorInit En_Encount3_InitVars = {
     (ActorFunc)EnEncount3_Draw,
 };
 
-#endif
-
-extern UNK_TYPE D_060009A0;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/EnEncount3_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/EnEncount3_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/func_809AD058.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/func_809AD084.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/func_809AD194.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/func_809AD1EC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/EnEncount3_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/EnEncount3_Draw.s")
+s32 D_809AD810 = false;
+
+void EnEncount3_Init(Actor* thisx, PlayState* play) {
+    EnEncount3* this = THIS;
+
+    this->unk14A = ENCOUNT3_GET_SPAWN_INDEX(thisx);
+    this->childParams = ENCOUNT3_GET_PARAM_F80(thisx);
+    this->switchFlags = ENCOUNT3_GET_SWITCHFLAGS(thisx);
+    this->unk15A = this->actor.csId;
+    this->unk16C = this->actor.world.rot.z * 10.0f;
```

Comment:
```suggestion
    this->unk16C = ENCOUNT3_GET_ROTZ(thisx) * 10.0f;
```
Ah wait, this project also wants Macros for parameters that come from un-used rotation input.

`Actor->params` is only s16 (16 bits) so some actors that want more parameter data will use their unused z and x rotation (most actors stand upright and don't need these) as a second and third location to pass parameter data for an actor, so `this->actor.world.rot.z` should get wrapped in its own parameter macro like 
```c
#define ENCOUNT3_GET_ROTZ(thisx) ((thisx)->world.rot.z)
```
Leaving the temporary name `GET_ROTZ`  is fine until someone figures out what it does. It _looks_ like it gets used to determine the activation range of the garo mask, but it might be used for a second purpose in a later function, so fine to leave it un-documented until it's explored more closely.

If you have to make the macro use `this` instead of `thisx` that's also fine.